dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoriesResource.java

```diff
@@ -70,4 +109,69 @@ public final Response getCategories(@Context final HttpServletRequest httpReques
 
         return response;
     }
+
+    /**
+     * Saves a new working version of a category.
+     *
+     * @param request
+     * @param response
+     * @param categoryForm
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @POST
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response saveNew(@Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            final CategoryForm categoryForm) throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requestAndResponse(request, response).rejectWhenNoUser(true).init();
+        final User user = initData.getUser();
+        final Host host = this.categoryHelper.getHost(categoryForm.getSiteId(),
+                () -> this.hostWebAPI.getCurrentHostNoThrow(request));
+        final PageMode pageMode = PageMode.get(request);
+
+        DotPreconditions.checkArgument(UtilMethods.isSet(categoryForm.getCategoryName()),
+                "The category name is required");
+
+        return Response.ok(new ResponseEntityView(this.categoryHelper.toCategoryView(
+                        this.fillAndSave(categoryForm, user, host, pageMode, new Category()), user)))
+                .build();
+    }
+
+    @WrapInTransaction
```

Comment:
We only use this annotation at java api level - it's not needed here. Check if categoryAPI.save has it.

Reply:
Yes at CategoryAPI we have @WrapInTransaction annotation already applied so I am removing it from the CategoryResource. Thanks for the feedback :)